source/settings/core.rst

```diff
@@ -2520,13 +2520,73 @@ Example Setting:
 
 A space-separated list of trusted network ranges.
 
-Typically used for IMAP proxy servers, allowing them to
-report a different IP address and port as the connection's origin for
-purposes of logging and authentication checks.
+This setting is used for a few different purposes, but most importantly it allows the client connection to tell the server what the original client's IP address was.
+This original client IP address is then used for logging and authentication checks.
 
-The ``disable_plaintext_auth setting`` is ignored for these networks.
+Plaintext authentication is always allowed for trusted networks (:ref:`setting-disable_plaintext_auth` is ignored).
 
-See :ref:`setting-disable_plaintext_auth`
+The details of how this setting works depends on the used protocol:
+
+IMAP:
+
+ * ID command can be used to override:
+
+   * Session ID
+   * Client IP and port (``%{rip}``, ``%{rport}``)
+   * Server IP and port (``%{lip}``, ``%{lport}``)
+
+ * ``forward_*`` fields can be sent to auth process's passdb lookup
+ * The trust is always checked against the connecting IP address.
+   Except if HAProxy is used, then the original client IP address is used.
+
+POP3:
+
+ * XCLIENT command can be used to override:
+
+   * Session ID
+   * Client IP and port (``%{rip}``, ``%{rport}``)
+
+ * ``forward_*`` fields can be sent to auth process's passdb lookup
+ * The trust is always checked against the connecting IP address.
+   Except if HAProxy is used, then the original client IP address is used.
+
+ManageSieve:
+
+ * XCLIENT command can be used to override:
+
+   * Session ID
+   * Client IP and port (``%{rip}``, ``%{rport}``)
+
+ * The trust is always checked against the connecting IP address.
+   Except if HAProxy is used, then the original client IP address is used.
+
+Submission:
+
+ * XCLIENT command can be used to override:
+
+   * Session ID
+   * Client IP and port (``%{rip}``, ``%{rport}``)
+   * HELO - FIXME?
+   * LOGIN - FIXME?
+   * PROTO - FIXME?
+
+ * ``forward_*`` fields can be sent to auth process's passdb lookup
+ * The trust is always checked against the connecting IP address.
+   Except if HAProxy is used, then the original client IP address is used.
+
+LMTP:
+
+ * XCLIENT command can be used to override:
+
+   * Session ID
+   * Client IP and port (``%{rip}``, ``%{rport}``)
+   * HELO - FIXME?
+   * LOGIN - FIXME?
+   * PROTO - FIXME?
```

Comment:
@stephanbosch what should I put to these FIXMEs? Do they apply to LMTP and submission both?

Reply:
HELO overrides what the client sent earlier in the EHLO/LHLO command. That value isn't currently of much significance. It is mentioned in the trace header if added and forwarded to the LMTP and Submission backends via XCLIENT It can also be sent to the Submission relay MTA if that supports XCLIENT.

LOGIN and PROTO don't seem to be used currently.